bindings/matrix-sdk-ffi/src/room.rs

```diff
@@ -265,6 +266,56 @@ impl Room {
         }
     }
 
+    pub async fn room_info(&self) -> Result<RoomInfo, ClientError> {
+        // Look for a local event in the `Timeline`.
+        //
+        // First off, let's see if a `Timeline` exists…
+        if let Some(timeline) = self.timeline.read().await.clone() {
+            // If it contains a `latest_event`…
+            if let Some(timeline_last_event) = timeline.latest_event().await {
+                // If it's a local echo…
+                if timeline_last_event.is_local_echo() {
+                    return Ok(RoomInfo::new(
+                        &self.inner,
+                        Some(Arc::new(EventTimelineItem(timeline_last_event))),
+                    )
+                    .await?);
+                }
+            }
+        }
+
+        // Otherwise, fallback to the classical path.
+        let latest_event = match self.inner.latest_event() {
```

Comment:
This code is duplicated from [`matrix_sdk_ui::room_list_service::Room::latest_event`](https://github.com/matrix-org/matrix-rust-sdk/blob/7aa6981e372dac0e3af78a629a34f90e66a24c07/crates/matrix-sdk-ui/src/room_list_service/room.rs#L147-L161). Don't you believe we can refactor to get this code in a single-place?